File: c#/classes/StaticClassVariable.cs
class Token {
  public static int Number;
  public string Color;
}

class StaticClassVariable {
  public static void Main(string[] args) {
    Token token_1 = new Token();
    Token token_2 = new Token();
    Token token_3 = new Token();

    System.Console.WriteLine("Token.Number: " + Token.Number + "\n");

    token_1.Color = "blue";
    token_2.Color = "orange";
    token_3.Color = "violet";
    Token.Number = 3;

    System.Console.WriteLine("token_1.Color: " + token_1.Color);
    System.Console.WriteLine("token_2.Color: " + token_2.Color);
    System.Console.WriteLine("token_3.Color: " + token_3.Color);

    System.Console.WriteLine("\nToken.Number: " + Token.Number + "\n");
  }
}

File: c++/classes/static_class_variable.cpp
#include <iostream>
#include <string>

class Token {
public:
  static int number;
  std::string color;
};

int Token::number;

int main() {
  std::cout << "Token::number: " << Token::number << std::endl << std::endl;

  Token token_1, token_2, token_3;

  token_1.number = 1;
  token_1.color = "blue";
  token_2.number = 2;
  token_2.color = "orange";
  token_3.number = 3;
  token_3.color = "violet";

  std::cout << "token_1.number: " << token_1.number << ", token_1.color: " << token_1.color << std::endl;
  std::cout << "token_2.number: " << token_2.number << ", token_2.color: " << token_2.color << std::endl;
  std::cout << "token_3.number: " << token_3.number << ", token_3.color: " << token_3.color << std::endl;

  std::cout << std::endl << "Token::number: " << Token::number << std::endl << std::endl;

  token_1.number = 1;
  token_1.color = "petrol";

  std::cout << "token_1.number: " << token_1.number << ", token_1.color: " << token_1.color << std::endl;
  std::cout << "token_2.number: " << token_2.number << ", token_2.color: " << token_2.color << std::endl;
  std::cout << "token_3.number: " << token_3.number << ", token_3.color: " << token_3.color << std::endl;

  std::cout << std::endl << "Token::number: " << Token::number << std::endl << std::endl;

  token_2.number = 2;
  token_2.color = "ginger";

  std::cout << "token_1.number: " << token_1.number << ", token_1.color: " << token_1.color << std::endl;
  std::cout << "token_2.number: " << token_2.number << ", token_2.color: " << token_2.color << std::endl;
  std::cout << "token_3.number: " << token_3.number << ", token_3.color: " << token_3.color << std::endl;

  std::cout << std::endl << "Token::number: " << Token::number << std::endl << std::endl;
}
